src/test/java/com/shapesecurity/shift/es2017/parser/miscellaneous/AsyncAwaitTest.java

```diff
@@ -131,5 +131,9 @@ public void testAsyncFailure() {
 		testScriptFailure("async (a = (await) => {}) => {}", 12, "Async arrow parameters may not contain \"await\"");
 		testScriptFailure("async (a = aw\\u{61}it => {}) => {}", 11, "Async arrow parameters may not contain \"await\"");
 		testScriptFailure("async (a = (b = await (0)) => {}) => {}", 16, "Async arrow parameters may not contain \"await\"");
+		testScriptFailure("async ({await}) => 1",  13,"Async arrow parameters may not contain \"await\"");
```

Comment:
Should add similar tests for `yield`, since this PR is adding those cases.

Reply:
I guess the only missing ones are `function *a({yield}){}` and `function *a({yield = 0}){}`.